packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -101,16 +126,60 @@ export default function MongoNav({
   admin = false,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
+
+  const accountHost = hosts?.account ?? `https://account.mongodb.com`;
+  const cloudHost = hosts?.cloud ?? `https://cloud.mongodb.com`;
+  const universityHost = hosts?.university ?? `https://university.mongodb.com`;
+  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  const endpointURI = `${cloudHost}/user/shared`;
+
+  function getProductionData() {
+    return fetch(endpointURI, {
+      credentials: 'include',
+      method: 'GET',
+      mode: 'cors',
+    });
+  }
+
+  function getFixtureData() {
+    return new Promise(resolve => {
+      setTimeout(() => {
+        onSuccess?.(fixtureData);
+
+        resolve(fixtureData);
+      }, 10);
+    });
+  }
+
+  async function handleResponse(response: Response) {
+    if (!response.ok) {
+      onError?.(ErrorCodeMap[response.status as 401]); //typecasting for now until we have more types to handle
```

Comment:
can do this: `const status = response.status as 401; //typecasting for now until we have more types to handle` but can't tell it that a response.status will always be 401 because TS knows it can be any number. Right now, we're only handling 401 which is why we have to cast it